healsparse/healSparseCoverage.py

```diff
@@ -75,6 +76,8 @@ def make_empty(cls, nside_coverage, nside_sparse):
         healSparseCoverage : `HealSparseCoverage`
            HealSparseCoverage from file
         """
+        if nside_coverage >= 128:
+            warnings.warn('Using `nside_coverage` >= 128 may allocate large arrays in memory!')
```

Comment:
First of all, I'm not 100% sure if 128 is the right nside to do a warning.  I think we might need to benchmark it.  However, anecdotally I can say that I have found nside=128 useful in some contexts (and not overwhelming), so these warnings should be `> 128` and not `>= 128`.  I also wonder if we should have a flag to suppress the warning if the user knows what they're doing (or they can always wrap in a `warnings` context, which is more flexible).
Also, I think this should be a `ResourceWarning` which is "for warnings related to resource usage."  Which I think this is, but maybe `UserWarning` is appropriate?  This I'm less sure of.

Reply:
The other problem, not just the large arrays, is that the performance is going to suffer if nside coverage gets too large.